integration_tests/src/main/python/aqe_test.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+from pyspark.sql.functions import when, col
+from pyspark.sql.types import *
+from asserts import assert_gpu_and_cpu_are_equal_collect
+from data_gen import *
+from marks import ignore_order, allow_non_gpu
+from spark_session import with_cpu_session
+
+_adaptive_conf = { "spark.sql.adaptive.enabled": "true" }
+
+# Dynamic switching of join strategies
+# Dynamic coalescing of shuffle partitions
+# Dynamically Handle Skew Joins
+
+_adaptive_coalese_conf = copy_and_update(_adaptive_conf, {
+    "spark.sql.adaptive.coalescePartitions.enabled": "true"
+})
+
+
+def create_skew_df(spark, length):
+    root = spark.range(0, length)
+    mid = length / 2
+    left = root.select(
+        when(col('id') < mid / 2, mid).
+            otherwise('id').alias("key1"),
+        col('id').alias("value1")
+    )
+    right = root.select(
+        when(col('id') < mid, mid).
+            otherwise('id').alias("key2"),
+        col('id').alias("value2")
+    )
+    return left, right
+
+
+@ignore_order(local=True)
+def test_aqe_skew_join():
+    def do_join(spark):
+        left, right = create_skew_df(spark, 500)
```

Comment:
is there anyway for us to really know AQE kicked in here?  The different versions of Spark have changed things so I can see some of these being different or potentially changing and not testing what we meant.  Perhaps we need to enhance our plan checks to make sure AQE is there and it did what we expect.  I'm ok if that is follow on if we need

Reply:
so for many of these tests, it's not as much a question of whether AQE kicked. In Databricks, these are each basically calling out the leaf exec's where computeStats needs to be implemented. That is only called when AQE is enabled on Databricks. I could see them backporting this to future Apache versions though. There is already a scala test version of this one that actually examines the plan. Perhaps I could look into how that one is kicked in?

Reply:
ok lets just put comments on the ones that are testing computeStats.  

we can do the testing separately with https://github.com/NVIDIA/spark-rapids/issues/6253